dsp/transform/hilbert_example_test.go

```diff
@@ -0,0 +1,46 @@
+package transform
+
+import (
+	"fmt"
+	"math/cmplx"
+)
+
+func ExampleHilbert_AnalyticSignal() {
+	// Samples is a set of real amplitudes that make up a signal
+	samples := []float64{1, 0, 2, 0, 4, 0, 2, 0}
+
+	// Initialize a Hilbert transform and 'demodulate' to get the analytic signal.
+	// The result is the complex I/Q (In-Phase / Quadrature) demodulation of the input signal.
+	h := NewHilbert(len(samples))
+	iqSamples := h.AnalyticSignal(nil, samples)
+
+	// We can compute the instantaneous amplitude of the signal (or 'envelope') using absolute value.
+	// Analyzing the envelope is an easy way to measure changes in amplitude over time in a signal.
+	envelope := make([]float64, len(samples))
+	for ind, iq := range iqSamples {
+		envelope[ind] = cmplx.Abs(iq)
+	}
+
+	// We can also compute the instantaneous phase of each part of the signal using the 4-quadrant arc-tangent.
+	// With multiple samples, the instantaneous phase can be used to estimate instantaneous frequency of a signal.
+	phase := make([]float64, len(samples))
+	for ind, iq := range iqSamples {
+		phase[ind] = cmplx.Phase(iq)
+	}
+
+	for i, iq := range iqSamples {
+		fmt.Printf("ind=%d -> I=%.4g, Q=%.4g, envelope=%.4g, phase=%.4g\n",
```

Comment:
```suggestion
		fmt.Printf("ind=%d -> I=%.4f, Q=%.4f, envelope=%.4f, phase=%.4f\n",
```
`%g` offers the opportunity to represent ~0 as the tiny value, which we probably don't want.